### PR TITLE
Groot new tool and data manager

### PR DESCRIPTION
**Pull request by hugolefeuvre**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [x] This PR adds a new tool or tool collection
* [ ] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

### CODE REVIEW

File: tools/groot/test-data/resfinder.90/cluster-1.msa [hunk 1]
```diff
@@ -0,0 +1,41 @@
+>*ARR-2_1_HQ141279
```
Comment (bernt-matthias) [inline tools/groot/test-data/resfinder.90/cluster-1.msa:1]: Seems  the `resfinder.90` directory is not used in the tests. If so it should be removed.

Comment (hugolefeuvre) [reply tools/groot/test-data/resfinder.90/cluster-1.msa:1]: It's used in the groot_align.xml test : `<param name="groot_db_select" value="resfinder.90" />`

File: data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<tool id="groot_database_downloader" name="Groot get" tool_type="manage_data" version="0.0.5" profile="23.0">
+    <description> Download pre-clustered ARG database that are ready to be indexed</description>
+    <requirements>
+        <requirement type="package" version="1.1.2">groot</requirement>
```
Comment (bebatut) [inline data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml:4]:
```suggestion
        <requirement type="package" version="3.12">python</requirement>
        <requirement type="package" version="1.1.2">groot</requirement>
```

File: data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<tool id="groot_database_downloader" name="Groot get" tool_type="manage_data" version="0.0.5" profile="23.0">
+    <description> Download pre-clustered ARG database that are ready to be indexed</description>
+    <requirements>
+        <requirement type="package" version="3.12">python</requirement>
+        <requirement type="package" version="1.1.2">groot</requirement>
```
Comment (bebatut) [inline data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml:5]:
```suggestion
        <requirement type="package" version="2024.6.2">ca-certificates</requirement>
        <requirement type="package" version="1.1.2">groot</requirement>
```

File: tools/groot/groot.xml [hunk 4]
```diff
@@ -0,0 +1,166 @@
+<tool id="groot_align" name="Groot align" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="@PROFILE@">
+    <description>reads to references and weight variation graphs and generate report</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="xrefs"/>
+    <expand macro="requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+ln -s '$fastq' 'input.fastq' &&
```
Comment (bernt-matthias) [inline tools/groot/groot.xml:9]: We should add `set -x pipefail;` if pipes are used, or?

File: tools/groot/groot.xml [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    #if $report.arg_report.criteria == 'cutoff'
+    --covCutoff $report.arg_report.covCutoff
+    #else
+    --lowCov
+    #end if
+    > '$report_out'
+
+]]></command>
+    <inputs>
+        <param argument="--fastq" type="data" format="fastq,fastqsanger,fastq.gz,fastqsanger.gz" label="FASTQ file(s) to align"/>
```
Comment (bernt-matthias) [inline tools/groot/groot.xml:42]: Please remove `fastq` and `fastq.gz` and list the supported types explicitly.

File: tools/groot/groot.xml [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    --fastq 'input.${extension}'
+    --indexDir 'grootIndex'
+    --contThresh $contThresh
+    --minKmerCov $minKmerCov
+    --log $log
+    $noAlign
+    --processors "\${GALAXY_SLOTS:-1}"
+
+| groot report 
+    #if $report.arg_report.criteria == 'cutoff'
```
Comment (bernt-matthias) [inline tools/groot/groot.xml:33]: Is it (or should it) be possible to call wo `--covCutoff` and `--lowCov`? Maybe we can also remove the conditional?

File: data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.py [hunk 7]
```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python
```
Comment (bernt-matthias) [inline data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.py:1]: May I suggest to drop all this python stuff and simplify this along  the lines of https://github.com/Helmholtz-UFZ/galaxy-tools/blob/main/data_managers/data_manager_checkv/data_manager/checkv_datamanager.xml

File: data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        <requirement type="package" version="3.12">python</requirement>
+        <requirement type="package" version="2024.6.2">ca-certificates</requirement>
+        <requirement type="package" version="1.1.2">groot</requirement>
+    </requirements>
+    <command><![CDATA[
+if [ -f '/usr/local/ssl/cacert.pem' ] ; then export SSL_CERT_FILE='/usr/local/ssl/cacert.pem' ; fi &&
+
+python '$__tool_directory__/data_manager_groot_database_downloader.py' 
+        --database '$database'
+        --grootversion 1.1.2
```
Comment (bernt-matthias) [inline data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml:13]: The version should be controlled by a macro token.

File: data_managers/data_manager_groot_database_downloader/tool_data_table_conf.xml.sample [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<tables>
+    <table name="groot_database" comment_char="#">
+        <columns>value, name, dbkey, path, db_version</columns>
```
Comment (bernt-matthias) [inline data_managers/data_manager_groot_database_downloader/tool_data_table_conf.xml.sample:3]: Should we also collect the groot version that has been used to download the data?

File: tools/groot/groot.xml (hunk 4 above)
Comment (hugolefeuvre) [reply tools/groot/groot.xml:9]: Yes, in top of the command part ?

File: tools/groot/groot.xml (hunk 5 above)
Comment (bernt-matthias) [reply tools/groot/groot.xml:42]: fastq covers all `fastq*` datatypes (`fastq = {fastqsanger,fastqillumina,fastqcsanger,...}`) which we probably do not want. I think its better to list all supported FASTQ variants.

File: tools/groot/groot.xml (hunk 6 above)
Comment (hugolefeuvre) [reply tools/groot/groot.xml:33]: I'm not sure is it possible to call without both options, in local if we don't specify any option, covCutoff is use by default.
Then for the conditionnal, --lowCov: overrides covCutoff option so I'm not sure we can remove it...

Comment (bernt-matthias) [reply tools/groot/groot.xml:33]: Was just a suggestion/question.

Conditionals have at the moment the downside that they can not be changed by the user in workflows.

File: data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+     <configfiles>
+        <configfile name="dmjson"><![CDATA[
+#from datetime import date
+{
+  "data_tables":{
+    "groot_database":[
+      {
+        "value": "$database.$identity",
+        "name": "$database ($identity percent identity)",
+        "path": "/$database.$identity",
```
Comment (bernt-matthias) [inline data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml:26]:
```suggestion
        "path": "$database",
```

Comment (bernt-matthias) [reply data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml]: Or you need to `-d '${database}.${identity}'` and use `"path": "$database.$identity",` here (also wo the slash).

File: data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml [hunk 11]
```diff
@@ -0,0 +1,70 @@
+<tool id="groot_database_downloader" name="Groot get" tool_type="manage_data" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="@PROFILE@">
+    <description> Download pre-clustered ARG database that are ready to be indexed</description>
+    <macros>
+        <token name="@TOOL_VERSION@">1.1.2</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+        <token name="@PROFILE@">22.05</token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="3.12">python</requirement>
+        <requirement type="package" version="2024.6.2">ca-certificates</requirement>
```
Comment (bernt-matthias) [inline data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml:10]:
```suggestion
```

File: data_managers/data_manager_groot_database_downloader/data_manager_conf.xml [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+        <data_table name="groot_database">
+            <output>
+                <column name="value"/>  <!-- columns that are going to be specified by the Data Manager Tool -->
+                <column name="name"/>  <!-- columns that are going to be specified by the Data Manager Tool -->
+                <column name="path" output_ref="out_file">
+                    <move type="directory">
+                        <source>${path}</source>
+                        <target base="${GALAXY_DATA_MANAGER_DATA_PATH}">groot_database/data/${value}</target>
+                    </move>
+                    <value_translation>${GALAXY_DATA_MANAGER_DATA_PATH}/groot_database/data/${value}</value_translation>
```
Comment (bernt-matthias) [inline data_managers/data_manager_groot_database_downloader/data_manager_conf.xml:13]: Is the `data` component in the path needed?

```suggestion
                    <move type="directory">
                        <source>${path}</source>
                        <target base="${GALAXY_DATA_MANAGER_DATA_PATH}">groot_database/${value}</target>
                    </move>
                    <value_translation>${GALAXY_DATA_MANAGER_DATA_PATH}/groot_database/${value}</value_translation>
```

File: tools/groot/groot.xml [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+                </assert_contents>
+            </output>
+        </test>
+    </tests>
+    <help><![CDATA[
+**What it does**
+
+@HELP@
+
+The align subcommand is used to align reads against the indexed variation graphs.
```
Comment (bernt-matthias) [inline tools/groot/groot.xml:147]: Are the more subcommands? Then we should make this a suite, i.e. slighly adapt the shed.yaml file

Comment (hugolefeuvre) [reply tools/groot/groot.xml:147]: Yes, there was several subcommands at first but we group them in a unique xml (index, align and report), I'm gonna change the help part

**Comment by hugolefeuvre** [comment]
PR prête à être merger, j'ai tenté de condenser le DM mais lorsque je passe uniquement avec un DM en xml, le test passe en local mais pas sur le git, erreur : 
```
downloading the pre-clustered arg-annot database...
could not download the tarball
Get "https://github.com/will-rowe/groot/raw/master/db/clustered-ARG-databases/1.1/arg-annot.90.tar": tls: failed to verify certificate: x509: certificate signed by unknown authority
```
Je suis donc revenu à un DM avec un fichier python, cette version fonctionne en local et sur git.

**Comment by bernt-matthias** [comment]
For fixing the certificate error you need the line `if [ -f '/usr/local/ssl/cacert.pem' ] ; then export SSL_CERT_FILE='/usr/local/ssl/cacert.pem' ; fi &&` in the command and the requirement `<requirement type="package" version="2024.6.2">ca-certificates</requirement>`. Then the nun-python version _should_ work...

**Comment by hugolefeuvre** [comment]
> For fixing the certificate error you need the line `if [ -f '/usr/local/ssl/cacert.pem' ] ; then export SSL_CERT_FILE='/usr/local/ssl/cacert.pem' ; fi &&` in the command and the requirement `<requirement type="package" version="2024.6.2">ca-certificates</requirement>`. Then the nun-python version _should_ work...

Indeed...I just had to not delete the line in the command part when I changed from the python version to the non-python version... Thank you

File: data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+}]]>
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param name="database" type="select" label="Database name">
+            <option value="arg-annot" selected="true">ARG-annot 90% identity (default)</option>
+            <option value="resfinder">Resfinder 90% identity</option>
+            <option value="card">CARD 90% identity</option>
+            <option value="groot-db">Groot-db 90% identity, combine all sequences in ARG-annot, Resfinder and CARD</option>
+            <option value="groot-core-db">Groot-core-db 90% identity, same as groot-db but one copy of each sequence is kept and then this collection is clustered at 90% identity</option>
```
Comment (bebatut) [inline data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml:41]:
```suggestion
            <option value="arg-annot" selected="true">ARG-annot</option>
            <option value="resfinder">Resfinder</option>
            <option value="card">CARD</option>
            <option value="groot-db">Groot-db (combine sequences in ARG-annot, Resfinder and CARD)</option>
            <option value="groot-core-db">Groot-core-db (same as groot-db but one copy of each sequence is kept and then this collection is clustered at given percentage identity)</option>
```

File: data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+         </param>
+         <param name="identity" type="select" label="Percent Identity">
+            <option value="90" selected="true">90% identity (default)</option>
```
Comment (bebatut) [inline data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml:44]:
```suggestion
            <option value="90" selected="true">90% identity</option>
```

File: data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+                    <has_text_matching expression='"value": "arg-annot.90-1.1.2-[0-9]{8}"'/>
+                    <has_text text='"name": "arg-annot (90 percent identity)"'/>
+                    <has_text text='"path": "/arg-annot.90"'/>
+                </assert_contents>
+            </output>
+        </test>
+    </tests>
+    <help>
+        The groot get subcommand is used to download a pre-clustered ARG database that is ready to be indexed.
+    </help>
```
Comment (bebatut) [inline data_managers/data_manager_groot_database_downloader/data_manager/data_manager_groot_database_downloader.xml:67]:
```suggestion
    <help><![CDATA[
The groot get subcommand is used to download a pre-clustered ARG database that is ready to be indexed.
]]></help>
```

File: tools/groot/groot.xml [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+    --numPart $numPart
+    --sketchSize $sketchSize
+&&
+
+groot align
+    --fastq 'input.${extension}'
+    --indexDir 'grootIndex'
+    --contThresh $contThresh
+    --minKmerCov $minKmerCov
+    --log $log
```
Comment (bebatut) [inline tools/groot/groot.xml:28]:
```suggestion
    --log '$log'
```

**Review by bebatut — APPROVED** [review]
Looks great for me. Thanks a lot @hugolefeuvre!!!

**Comment by mvdbeek** [comment]
Attention: deployment failure!

https://github.com/galaxyproject/tools-iuc/actions/runs/11386661801

**Comment by mvdbeek** [comment]
Attention: deployment failure!

https://github.com/galaxyproject/tools-iuc/actions/runs/11386661801

**Comment by bernt-matthias** [comment]
You can open a PR introducing any change in on of the tools of the suite.

**Pull request merged**